lib/default.nix

```diff
@@ -11,6 +11,14 @@ let
     callLibs = file: import file { lib = self; };
   in {
 
+    # Configuration of lib, can be set by doing
+    #   lib.extend (self: super: { config = ...; })
+    # One can get this value through `lib.config`. The Nixpkgs configuration
+    # will be passed like this.
+    config = {
+      nameSourcesPrettily = false;
```

Comment:
If at all this should be in a separate PR with appropriate justification, tests and documentation.
And I'm also not a big fan of this because it makes `lib` functions impure in a way: `lib.foo` can change behavior based on `config` without taking it as an argument.